src/formatDate.js

```diff
@@ -8,7 +8,38 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat[fromFormat.length - 1];
+  const toSeparator = toFormat[toFormat.length - 1];
+  const fromParts = fromFormat.slice(0, -1);
+  const toParts = toFormat.slice(0, -1);
+
+  const dateParts = date.split(fromSeparator);
+
+  const dateObj = {};
+
+  fromParts.forEach((part, index) => {
+    dateObj[part] = dateParts[index];
+  });
+
+  const formattedDate = toParts
+    .map((part) => {
+      if (part === 'YYYY' && dateObj['YY']) {
+        const year = parseInt(dateObj['YY'], 10);
+
+        if (year < 30) {
+          return '20' + dateObj['YY'];
+        } else {
+          return '19' + dateObj['YY'];
```

Comment:
When converting from 'YY' to 'YYYY', it is assumed that any 'YY' less than 30 belongs to the 21st century, and any 'YY' 30 or above belongs to the 20th century. This is a reasonable assumption for current dates but may not hold true in all contexts. If this logic is part of the requirements, ensure that it's clearly documented in the function description.